### PR TITLE
[bitnami/postgresql] Add metrics support to read replica

### LINKED ISSUE #9869: Add metrics to read replicas
### Name and Version

bitnami/postgresql:11.1.22

### What is the problem this feature will solve?

Metrics are not available for read replica.

### What is the feature you are proposing to solve the problem?

Add the following to the read replica templates:

> Metrics sidecar to read statefulset.
> Metrics configmap for read replica.
> ServiceMonitor for read replica.
> PrometheusRule CRD for both read and primary replicas to avoid double alerting rules.

### What alternatives have you considered?

_No response_

### PR DESCRIPTION
**Pull request by ajammali**
Signed-off-by: Anouar Jammali <jammali.anouar@gmail.com>

**Description of the change**
Add the following to the read replica templates:

> Metrics sidecar to read statefulset.
> Metrics configmap for read replica.
> ServiceMonitor for read replica.
> PrometheusRule CRD for both read and primary replicas to avoid double alerting rules.

**Benefits**

Generate metrics for the read replication.

**Possible drawbacks**

**Applicable issues**

  - fixes #9869 

**Additional information**

**Checklist**

- [ ] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [ ] Variables are documented in the values.yaml and added to the README.md using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [ ] Title of the PR starts with chart name (e.g. [bitnami/<name_of_the_chart>])
- [ ] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Review by dgomezleon — CHANGES_REQUESTED** [review]
Thanks a lot for contributing. I have just required some minor changes.

File: bitnami/postgresql/templates/prometheusrule.yaml [hunk 1]
```diff
@@ -2,8 +2,8 @@
 apiVersion: monitoring.coreos.com/v1
 kind: PrometheusRule
 metadata:
-  name: {{ include "postgresql.primary.fullname" . }}
-  namespace: {{ default .Release.Namespace .Values.metrics.prometheusRule.namespace | quote }}
+  name: {{ include "common.names.fullname" . }}
+  namespace: {{ default .Release.Namespace .Values.metrics.prometheusRule.namespace }}
```
[the remark this answers is not in the document]
Comment (ajammali) [reply bitnami/postgresql/templates/prometheusrule.yaml]: checked

File: bitnami/postgresql/templates/read/servicemonitor.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
+{{- if and .Values.metrics.enabled .Values.metrics.serviceMonitor.enabled }}
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: {{ include "postgresql.readReplica.fullname" . }}
+  namespace: {{ default .Release.Namespace .Values.metrics.serviceMonitor.namespace }}
```
[the remark this answers is not in the document]
Comment (ajammali) [reply bitnami/postgresql/templates/read/servicemonitor.yaml]: checked

File: bitnami/postgresql/templates/read/servicemonitor.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      {{- end }}
+      {{- if .Values.metrics.serviceMonitor.metricRelabelings }}
+      metricRelabelings: {{- include "common.tplvalues.render" ( dict "value" .Values.metrics.serviceMonitor.metricRelabelings "context" $) | nindent 6 }}
+      {{- end }}
+      {{- if .Values.metrics.serviceMonitor.honorLabels }}
+      honorLabels: {{ .Values.metrics.serviceMonitor.honorLabels }}
+      {{- end }}
+  namespaceSelector:
+    matchNames:
+      - {{ .Release.Namespace }}
```
[the remark this answers is not in the document]
Comment (ajammali) [reply bitnami/postgresql/templates/read/servicemonitor.yaml]: checked

File: bitnami/postgresql/templates/read/metrics-configmap.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+{{- if and .Values.metrics.enabled .Values.metrics.customMetrics }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ printf "%s-metrics" (include "postgresql.readReplica.fullname" .) }}
+  namespace: {{ .Release.Namespace }}
```
[the remark this answers is not in the document]
Comment (ajammali) [reply bitnami/postgresql/templates/read/metrics-configmap.yaml:6]: checked

[the remark this answers is not in the document]
Comment (dgomezleon) [reply bitnami/postgresql/templates/read/metrics-configmap.yaml:6]: It seems you missed this one.

**Comment by rafariossaa** [comment]
Also, take a look to the conflict in `Chart.yaml`

File: bitnami/postgresql/templates/read/metrics-configmap.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (ajammali) [reply bitnami/postgresql/templates/read/metrics-configmap.yaml]: done

**Review by dgomezleon — APPROVED** [review]
LGTM

**Pull request merged**